core/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -467,10 +468,19 @@ protected void doXContentAnalyzers(XContentBuilder builder, boolean includeDefau
             if (includeDefaults) {
                 builder.field("analyzer", "default");
             }
-        } else if (includeDefaults || fieldType().indexAnalyzer().name().startsWith("_") == false && fieldType().indexAnalyzer().name().equals("default") == false) {
-            builder.field("analyzer", fieldType().indexAnalyzer().name());
-            if (fieldType().searchAnalyzer().name().equals(fieldType().indexAnalyzer().name()) == false) {
+        } else {
+            boolean writtenSearchAnalyzer = false;
+            if (includeDefaults || fieldType().searchAnalyzer().name().startsWith("_") == false
+                    && fieldType().searchAnalyzer().name().equals("default") == false
+                    && fieldType().searchAnalyzer().name().equals(fieldType().indexAnalyzer().name()) == false) {
                 builder.field("search_analyzer", fieldType().searchAnalyzer().name());
+                writtenSearchAnalyzer = true;
```

Comment:
You can simplify this to:
```
boolean writeSearchAnalyzer =  // logic
if (writeSearchAnalyzer || analyzer logic) {
  // write analyzer
}
if (writeSearchAnalyzer) {
    // write search_analyzer
}
```

This will also keep the same order (analyzer followed by search_analyzer) that we had before.